data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/events/EventSubscriber.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.beaconrestapi.handlers.v1.events;
+
+import io.javalin.http.sse.SseClient;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import tech.pegasys.teku.infrastructure.async.AsyncRunner;
+
+public class EventSubscriber {
+  private static final Logger LOG = LogManager.getLogger();
+  private static final int MAX_PENDING_EVENTS = 10;
+  private final List<EventType> eventTypes;
+  private final SseClient sseClient;
+  private final Queue<QueuedEvent> queuedEvents;
+  private final Runnable closeCallback;
+  private final AtomicBoolean processingQueue;
+  final AsyncRunner asyncRunner;
+
+  public EventSubscriber(
+      final List<String> eventTypes,
+      final SseClient sseClient,
+      final Runnable closeCallback,
+      final AsyncRunner asyncRunner) {
+    this.eventTypes = getTopics(eventTypes);
+    this.sseClient = sseClient;
+    this.closeCallback = closeCallback;
+    this.queuedEvents = new ConcurrentLinkedQueue<>();
+    this.processingQueue = new AtomicBoolean(false);
+    this.asyncRunner = asyncRunner;
+    this.sseClient.onClose(closeCallback);
+  }
+
+  public void onEvent(final EventType eventType, final String message) {
+    if (eventTypes.contains(eventType)) {
```

Comment:
(nit) We could do this to avoid nesting:
```suggestion
    if (!eventTypes.contains(eventType)) {
      return;
    }
```